frontend/ui/elements/screensaver_menu.lua

```diff
@@ -0,0 +1,188 @@
+local Screensaver = require("ui/screensaver")
+local _ = require("gettext")
+
+local function screensaverType() return G_reader_settings:readSetting("screensaver_type") end
+local function screensaverDelay() return G_reader_settings:readSetting("screensaver_delay") end
+local function lastFile() return G_reader_settings:readSetting("lastfile") end
+local function messageBackground() return G_reader_settings:isTrue("message_background") end
+
+return {
+    {
+        text = _("Use last book's cover as screensaver"),
+        enabled_func = function() return lastFile() ~= nil end,
+        checked_func = function()
+            if screensaverType() == "cover" then
+                return true
+            else
+                return false
+            end
+        end,
+        callback = function()
+            G_reader_settings:saveSetting("screensaver_type", "cover")
+        end
+    },
+    {
+        text = _("Use book status as screensaver"),
+        enabled_func = function() return lastFile() ~= nil end,
+        checked_func = function()
+            if screensaverType() == "bookstatus" then
+                return true
+            else
+                return false
+            end
+        end,
+        callback = function()
+            G_reader_settings:saveSetting("screensaver_type", "bookstatus")
+        end
+    },
+    {
+        text = _("Use random image from folder as screensaver"),
+        checked_func = function()
+            if screensaverType() == "random_image" then
+                return true
+            else
+                return false
+            end
+        end,
+        callback = function()
+            G_reader_settings:saveSetting("screensaver_type", "random_image")
+        end
+    },
+    {
+        text = _("Use reading progress as screensaver"),
+        enabled_func = function() return Screensaver.getReaderProgress ~= nil and lastFile() ~= nil end,
+        checked_func = function()
+            if screensaverType() == "readingprogress" then
+                return true
+            else
+                return false
+            end
+        end,
+        callback = function()
+            G_reader_settings:saveSetting("screensaver_type", "readingprogress")
+        end
+    },
+    {
+        text = _("Use message as screensaver"),
+        checked_func = function()
+            if screensaverType() == "message" then
+                return true
+            else
+                return false
+            end
+        end,
+        callback = function()
+            G_reader_settings:saveSetting("screensaver_type", "message")
+        end
+    },
+    {
+        text = _("Disabled (let screen/page as it is)"),
```

Comment:
You *leave* something as it is. :-)

Reply:
Do you mean "Disabled (let screen/page)" or simply "Disabled"?

Reply:
I think he corrected the string I suggested :) It should be "Disabled (*leave* screen/page as it is)"

Reply:
Ok, I see now :)
Thanks :)

Reply:
Indeed, that's what I meant.

Perhaps instead of "disabled" something like "no action" might be clearer?

Reply:
Or just ``Leave screen/page as it is`` (the setting can stay named "disable").
"Disabled" may indeed convey the idea that suspend/battery saving would be disabled.

Reply:
Yes, I like that. `Leave screen as it is` should probably suffice?

Reply:
Agreed.